spec/rails_state_machine/state_machine/callback_spec.rb

```diff
@@ -0,0 +1,33 @@
+describe RailsStateMachine::StateMachine do
+
+  describe 'creating an object' do
+    # it is not intended to call an event directly on a new object e.g. Parcel.new(weigth: 1).pack! will not work
+  end
+
+  describe 'updating an object' do
+    subject(:parcel) { Parcel.create(weight: 1) }
+
+    before { parcel.callbacks.clear }
+
+    it 'runs all callbacks' do
+      expect { parcel.pack! }.to change { parcel.callbacks }.from([]).to([
+        'before_validation (standard)',
+        'before_validation (state machine)',
+        'before_save (standard)',
+        'before_save (state machine)',
+        'after_save (standard)',
+        'after_save (state machine)'
+      ])
+    end
+
+    it 'allows to make a state change in the after safe callback' do
+      expect { parcel.pack_and_ship! }.to change { parcel.reload.state }.from('empty').to('shipped')
+    end
+
+  end
+
+  describe 'destroying an object' do
+    # it is not intended to call an event which destroys the object
```

Comment:
What would happen when events destroy records? At first glance, it would work, wouldn't it? Saving a destroyed record actually "works" as in "will not raise an error".

I'd remove this empty block. Was your intention to keep it for documentation?